google-cloud-pub-sub-grpc/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/grpc/scaladsl/GooglePubSub.scala

```diff
@@ -47,27 +50,45 @@ object GooglePubSub {
   ): Source[ReceivedMessage, Future[Cancellable]] =
     Source
       .setup { (mat, attr) =>
-        val cancellable = Promise[Cancellable]
+        val sharedKillSwitch = KillSwitches.shared("subscribe")
 
         val subsequentRequest = request
           .withSubscription("")
           .withStreamAckDeadlineSeconds(0)
 
-        subscriber(mat, attr).client
-          .streamingPull(
-            Source
-              .single(request)
-              .concat(
-                Source
-                  .tick(0.seconds, pollInterval, ())
-                  .map(_ => subsequentRequest)
-                  .mapMaterializedValue(cancellable.success)
-              )
-          )
-          .mapConcat(_.receivedMessages.toVector)
-          .mapMaterializedValue(_ => cancellable.future)
+        val client = subscriber(mat, attr).client
+
+        def stream =
+          client
+            .streamingPull(
+              Source
+                .single(request)
+                .concat(
+                  Source
+                    .tick(0.seconds, pollInterval, ())
+                    .map(_ => subsequentRequest)
+                )
+            )
+            .mapConcat(_.receivedMessages.toVector)
+            .via(sharedKillSwitch.flow)
+
+        stream
+          .recoverWithRetries(-1, {
+            case e if StatusUtils.isRetryable(e) => stream
+          })
+          .mapMaterializedValue[Cancellable] { _ =>
```

Comment:
I'm not sure how `Cancellable` should be handled, I'm looking for some guidance.

Reply:
I'm open to breaking the API for 2.0.0 and change this to materialize a `KillSwitch` instead.
If we decide to do so, that should apply to `subscribePolling`.

#2169 in regular Pub/Sub is somewhat related and might make sense to apply to `subscribePolling` as well.

Reply:
It might be better to stay with using the tick cancel as this will allow draining the messages already received.
So you'd require an `AtomicReference` to the latest materialized `Cancellable` and expose that.